resources/benchmark-runner.mjs

```diff
@@ -147,57 +172,54 @@ export class BenchmarkRunner {
         }, 0);
     }
 
-    step(state) {
+    async step(state) {
         if (!state) {
             state = new BenchmarkState(this._suites);
             this._measuredValues = {tests: {}, total: 0, mean: NaN, geomean: NaN, score: NaN};
         }
 
         const suite = state.currentSuite();
         if (!suite) {
-            this._finalize();
-            return Promise.resolve();
+            await this._finalize();
+            return;
         }
 
         if (state.isFirstTest()) {
             this._removeFrame();
-            let self = this;
-            return state.prepareCurrentSuite(this, this._appendFrame()).then(prepareReturnValue => {
-                self._prepareReturnValue = prepareReturnValue;
-                return self._runTestAndRecordResults(state);
-            });
+            this._appendFrame();
+            this._page = new Page(this._frame);
+            await state.prepareCurrentSuite(this._frame, this._page);
```

Comment:
Why are we passing both this._frame and this._page?
Can't we just use pass page here?